src/main/kotlin/no/nav/k9/los/nyoppgavestyring/mottak/feltdefinisjon/FeltdefinisjonRepository.kt

```diff
@@ -33,15 +38,23 @@ class FeltdefinisjonRepository {
 
     fun fjern(sletteListe: Set<Feltdefinisjon>, tx: TransactionalSession) {
         sletteListe.forEach { datatype ->
-            tx.run(
-                queryOf(
-                    """
+            try {
+                tx.run(
+                    queryOf(
+                        """
                         delete from feltdefinisjon where id = :id""",
-                    mapOf(
-                        "id" to datatype.id,
-                    )
-                ).asUpdate
-            )
+                        mapOf(
+                            "id" to datatype.id,
+                        )
+                    ).asUpdate
+                )
+            } catch (e: PSQLException) {
+                if (e.sqlState.equals("23503")) {
+                    throw IllegalDeleteException("Kan ikke slette feltdefinisjon som brukes av oppgavetype", e)
```

Comment:
tenkte litt mer på dette. Kunne du skrevet om spørringen over til å ta hensin til at feltdefinisjonen ikke er i bruk av oppgavetype under slettingen - så slipper du hele exception problematikken.... (Kan også evt. gjøre det i neste omgang etter merge)

Reply:
Jeg er redd for at spørringen da blir så komplisert at man risikerer å innføre en eller annen bug og ikke avdekke feilen?

Det var en del av grunndesignet å bruke referanseintegriteten i databasen som sikkerhetsnett for modellen. 
Jeg ser ikke klart for meg hvordan den SQLen skulle sett ut. Måtte nok evt tatt det i neste iterasjon